### DISCUSSION TITLE
Configuring Vim Mode Switching Upon File Save

### DISCUSSION THREAD

**Discussion by delgermurun** [zed-industries/zed#8308] [Feature Requests]
Hey everyone,

I'm trying to get Zed to switch from insert mode to normal mode automatically after I save a file using "cmd-s". I've been looking around but can't seem to find how to set up a key binding that does more than one thing at a time.

Has anyone done something like this or knows how to? I'd really appreciate any tips or pointers!

Thanks a bunch!

**Comment by Moshyfawn** [comment]
I believe it was explored a few days ago in #7033. Have a look at https://github.com/zed-industries/zed/issues/7033#issuecomment-1955205499 and see if it fits your use case.

**Reply by delgermurun** [reply]
Thank you. ⬇️  worked for `Zed Preview 0.124.4`
```
{
    "context": "Editor && vim_mode == insert",
    "bindings": {
      "cmd-s": ["workspace::SendKeystrokes", "escape : w enter"]
    }
  }
```
